apps/homescreen/js/app.js

```diff
@@ -726,6 +748,97 @@
       return this.icons.children.indexOf(child);
     },
 
+    removeSelectedIcon: function() {
+      if (!this.selectedIcon) {
+        return;
+      }
+
+      if (this.selectedIcon.app) {
+        if (!this.selectedIcon.app.removable) {
+          return;
+        }
+        navigator.mozApps.mgmt.uninstall(this.selectedIcon.app);
+      } else if (this.selectedIcon.bookmark) {
```

Comment:
Just covering bases in case somehow and icon with no app or bookmark ends up here... Shouldn't be possible, but I'd rather fail gracefully. If we ever add a third category of selectable icon, it'll be easier to transition I guess too...